### PR TITLE
STONEBLD-2525 Move remote task generator

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This moves the remote task generator to the build-definitions repo. It means that changes to the generator can no go into the same commit as the corresponding remote task update.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,86 @@
+module github.com/konflux-ci/build-definitions
```
Comment (chmeliik) [inline go.mod:1]: Could you move the go.mod and go.sum to the remote-task-generator subdirectory?

**Review by chmeliik — COMMENTED** [review]
Nice, this will feel much better :taco:

File: go.mod (hunk 1 above)
Comment (zregvart) [reply go.mod:1]: :+1: Yeah, that would help with EC-690, move of tash for #1039 in here.

Comment (stuartwdouglas) [reply go.mod]: I might rename it to task-generator then, once this is in we can look at moving tash

**Comment by stuartwdouglas** [comment]
Should be ready now

**Pull request merged**